tsl/src/remote/connection_cache.c

```diff
@@ -156,49 +203,226 @@ connection_cache_create()
 	return cache;
 }
 
-Cache *
-remote_connection_cache_pin()
-{
-	return ts_cache_pin(connection_cache_current);
-}
-
 TSConnection *
-remote_connection_cache_get_connection(Cache *cache, TSConnectionId id)
+remote_connection_cache_get_connection(TSConnectionId id)
 {
 	CacheQuery query = { .data = &id };
-	ConnectionCacheEntry *entry = ts_cache_fetch(cache, &query);
+	ConnectionCacheEntry *entry = ts_cache_fetch(connection_cache, &query);
 
 	return entry->conn;
 }
 
-void
-remote_connection_cache_remove(Cache *cache, TSConnectionId id)
+bool
+remote_connection_cache_remove(TSConnectionId id)
 {
-	ts_cache_remove(cache, &id);
+	return ts_cache_remove(connection_cache, &id);
 }
 
 /*
  * Connection invalidation callback function
  *
  * After a change to a pg_foreign_server catalog entry,
- * mark the cache as invalid.
+ * mark the cache entry as invalid.
+ */
+void
+remote_connection_cache_invalidate_callback(Datum arg, int cacheid, uint32 hashvalue)
+{
+	HASH_SEQ_STATUS scan;
+	ConnectionCacheEntry *entry;
+
+	if (cacheid != FOREIGNSERVEROID)
```

Comment:
Assert?

Reply:
It's actually asserted in cache_invalidate_syscache_callback already so another assertion here wouldn't have any effect. Figured I'd leave this check here for clarity.